meteor/server/api/snapshot.ts

```diff
@@ -711,6 +756,86 @@ export async function restoreFromRundownPlaylistSnapshot(
 			snapshot.playlist.previousPartInstanceId
 	}
 
+	const expectedPackageIdMap = new Map<ExpectedPackageId, ExpectedPackageId>()
+	_.each(snapshot.expectedPackages, (expectedPackage) => {
+		const oldId = expectedPackage._id
+
+		let ownerId:
+			| PieceId
+			| AdLibActionId
+			| RundownBaselineAdLibActionId
+			| BucketAdLibId
+			| BucketAdLibActionId
+			| RundownId
+			| StudioId
+			| undefined = undefined
+
+		switch (expectedPackage.fromPieceType) {
+			case ExpectedPackageDBType.PIECE:
+			case ExpectedPackageDBType.ADLIB_PIECE:
+			case ExpectedPackageDBType.ADLIB_ACTION: {
+				expectedPackage.pieceId = pieceIdMap[unprotectString(expectedPackage.pieceId)]
+				expectedPackage.rundownId = rundownIdMap[unprotectString(expectedPackage.rundownId)]
+				expectedPackage.segmentId = segmentIdMap[unprotectString(expectedPackage.segmentId)]
+				ownerId = expectedPackage.pieceId
+				break
+			}
+			case ExpectedPackageDBType.BASELINE_ADLIB_PIECE:
+			case ExpectedPackageDBType.BASELINE_ADLIB_ACTION: {
+				expectedPackage.pieceId = pieceIdMap[unprotectString(expectedPackage.pieceId)]
+				expectedPackage.rundownId = rundownIdMap[unprotectString(expectedPackage.rundownId)]
+				ownerId = expectedPackage.pieceId
+				break
+			}
+			case ExpectedPackageDBType.RUNDOWN_BASELINE_OBJECTS: {
+				expectedPackage.rundownId = rundownIdMap[unprotectString(expectedPackage.rundownId)]
+				ownerId = expectedPackage.rundownId
+				break
+			}
+			case ExpectedPackageDBType.BUCKET_ADLIB:
```

Comment:
should we try to restore these? they don't belong to the rundown so could replace any newer version currently in the db. Same for STUDIO_BASELINE_OBJECTS